draco/tests/test_hard.py

```diff
@@ -319,3 +320,596 @@ def test_point_tick_bar_without_x_or_y():
+    attribute((scale,type),2,categorical).
+    attribute((scale,channel),3,y).
+    attribute((scale,type),3,ordinal).
+    """
+        )
+        == ["line_area_with_discrete"]
+    )
+
+
+def test_bar_tick_continuous_x_y():
+    b = hard.blocks["bar_tick_continuous_x_y"]
+    assert isinstance(b, Block)
+
+    assert no_violations(
+        b.program
+        + """
+    entity(mark,0,1).
+    attribute((mark,type),1,tick).
+    entity(encoding,1,2).
+    attribute((encoding,channel),2,x).
+    entity(scale,0,4).
+    attribute((scale,channel),4,x).
+    attribute((scale,type),4,linear).
+    """
+    )
+
+    assert no_violations(
+        b.program
+        + ("\n").join(
+            dict_to_facts(
+                {
+                    "mark": [
+                        {
+                            "type": "tick",
+                            "encoding": [
+                                {"channel": "x", "field": "temperature"},
+                                {"channel": "y", "field": "wind"},
+                            ],
+                        }
+                    ],
+                    "scale": [
+                        {"channel": "x", "type": "linear"},
+                        {"channel": "y", "type": "categorical"},
+                    ],
+                }
+            )
+        )
+    )
+
+    # multiple views
+    assert (
+        list_violations(
+            b.program
+            + ("\n").join(
+                dict_to_facts(
+                    {
+                        "view": [
+                            {
+                                "mark": [
+                                    {
+                                        "type": "tick",
+                                        "encoding": [
+                                            {"channel": "y", "field": "temperature"}
+                                        ],
+                                    }
+                                ],
+                                "scale": [{"channel": "y", "type": "linear"}],
+                            },
+                            {
+                                "mark": [
+                                    {
+                                        "type": "bar",
+                                        "encoding": [
+                                            {"channel": "x", "field": "temperature"},
+                                            {"channel": "y", "aggregate": "count"},
+                                        ],
+                                    }
+                                ],
+                                "scale": [
+                                    {"channel": "x", "type": "linear"},
+                                    {"channel": "y", "type": "linear"},
+                                ],
+                            },
+                        ]
+                    }
+                )
+            )
+        )
+        == ["bar_tick_continuous_x_y"]
```

Comment:
It's hard to see the error. Maybe add a comment where something is wrong similar to https://github.com/cmudig/draco2/blob/b6fba8c31bd2377a617b4a3aaf186734f9229607/draco/tests/test_spec_utils.py#L25